asm/sigreturn.S

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "asmdefs.h"
+
+//==============================================================================
+//
+// void myst_sigreturn(mcontext_t* mcontext)
+//
+//     Function used to return from the signal handler.
+//
+//     %rdi := mcontext
+//
+//==============================================================================
+
+.globl myst_sigreturn
+.type myst_sigreturn, @function
+myst_sigreturn:
+.cfi_startproc
+
+    // The fpregs member is a pointer
+    mov MYST_MCONTEXT_FPREGS(%rdi), %rsi
+    fxrstor (%rsi)
+
+    mov MYST_MCONTEXT_RSI(%rdi), %rsi
+    mov MYST_MCONTEXT_RDI(%rdi), %rdi
+
+    mov MYST_MCONTEXT_RAX(%rdi), %rax
+    mov MYST_MCONTEXT_RBX(%rdi), %rbx
+    mov MYST_MCONTEXT_RCX(%rdi), %rcx
+    mov MYST_MCONTEXT_RDX(%rdi), %rdx
+
+    mov MYST_MCONTEXT_R8(%rdi), %r8
+    mov MYST_MCONTEXT_R9(%rdi), %r9
+    mov MYST_MCONTEXT_R10(%rdi), %r10
+    mov MYST_MCONTEXT_R11(%rdi), %r11
+    mov MYST_MCONTEXT_R12(%rdi), %r12
+    mov MYST_MCONTEXT_R13(%rdi), %r13
+    mov MYST_MCONTEXT_R14(%rdi), %r14
+    mov MYST_MCONTEXT_R15(%rdi), %r15
+
+    mov MYST_MCONTEXT_RBP(%rdi), %rbp
+    mov MYST_MCONTEXT_RSP(%rdi), %rsp
+
+    sub $MYST_X86_64_ABI_REDZONE_SIZE, %rsp
+
+    pushq MYST_MCONTEXT_RIP(%rdi)
+
+    pushq MYST_MCONTEXT_EFL(%rdi)
```

Comment:
Intel SDK uses mitigation (notq) when setting up the flags register.
https://github.com/intel/linux-sgx/blob/e0cce77e910c4b265681597da4109f5e40f950b4/sdk/trts/linux/trts_pic.h#L124

It would be good to find out if we need this mitigation here. We can add the mitigation, and later remove it if it is not needed.

Reply:
It seems that this routine is for clearing the flags rather than restoring.